test/common/controllers/base-address.js

```diff
@@ -0,0 +1,154 @@
+    let baseAddressController;
+    const options = {
+      addressKey: 'foo'
+    };
+
+    beforeEach(()=> {
+      StubBaseController.prototype.locals = sinon.stub();
+      baseAddressController = new BaseAddressController(options);
+      baseAddressController.options = options;
+    });
+
+      describe('locals', () => {
+        let req = {
+          sessionModel: {
+            get: sinon.stub()
+          }
+        };
+        let res = {};
+
+        beforeEach(function() {
+          this.stub(BaseAddressController.prototype, 'mapAddress').returns([]);
+          this.stub(BaseAddressController.prototype, 'hasCategories').returns(false);
+        });
+
+        it('returns an empty address list if no addresses exist on the session', () => {
+          const expectedVal = {
+            hasAddresses: false,
+            hasCategories: false,
+            items: []
+          };
+          baseAddressController.locals(req, res).should.deep.equal(expectedVal);
+        });
+
+        it('calls base locals', () => {
+          StubBaseController.prototype.locals.should.not.have.been.called;
+
+          baseAddressController.locals(req, res);
+
+          StubBaseController.prototype.locals.should.have.been.called;
+        });
+
+        it('extends base.locals', () => {
+          const superLocalsVal = {foo: 'bar'};
+          StubBaseController.prototype.locals.returns(superLocalsVal);
+
+          baseAddressController.locals(req, res).should.have.property('foo')
+              .and.equal('bar');
+        });
+
+        it('includes the result of mapAddress as `items`', () => {
+          const mapped = [{id: 'abc123', address: '1 Some Road'}];
+          baseAddressController.mapAddress.returns(mapped);
+          const result = baseAddressController.locals(req, res);
+          result.should.have.property('items');
+          result.items.should.deep.equal(mapped);
+        });
+
+        it('includes the result of hasCategories', () => {
+          const categories = true;
+          baseAddressController.hasCategories.returns(categories);
+          const result = baseAddressController.locals(req, res);
+          result.should.have.property('hasCategories');
+          result.hasCategories.should.equal(categories);
+        });
+      });
+      describe('getAddresses', () => {
+        const fooValue = 'London';
+        let req = {
+          sessionModel: {
+            foo: fooValue,
+            get: (id) => req.sessionModel[id]
+          }
+        };
+        it('returns an expected address from the sessionModel', ()=> {
+          baseAddressController.getAddresses(req).should.equal(fooValue);
+        });
+
+        it('returns undefined if the options cannot be found in the sessionModel', () => {
+          const otherOptions = {
+            addressKey: 'blah'
+          };
+          baseAddressController.options = otherOptions;
+          should.equal(baseAddressController.getAddresses(req), undefined);
```

Comment:
I'd just add a `global.expect = chai.expect;` line to `test/common.js` and deal with the conflict later.